playbooks/cloud-provider-openstack-acceptance-test-lb-octavia/run.yaml

```diff
@@ -1,124 +1,138 @@
-          '{{ kubectl }}' config use-context local
-          # Hack for RBAC for all for the new cloud-controller process, we need to do better than this
-          '{{ kubectl }}' create clusterrolebinding --user system:serviceaccount:kube-system:default kube-system-cluster-admin-1 --clusterrole cluster-admin
-          '{{ kubectl }}' create clusterrolebinding --user system:serviceaccount:kube-system:pvl-controller kube-system-cluster-admin-2 --clusterrole cluster-admin
-          '{{ kubectl }}' create clusterrolebinding --user system:serviceaccount:kube-system:cloud-node-controller kube-system-cluster-admin-3 --clusterrole cluster-admin
-          '{{ kubectl }}' create clusterrolebinding --user system:serviceaccount:kube-system:cloud-controller-manager kube-system-cluster-admin-4 --clusterrole cluster-admin
-          '{{ kubectl }}' create clusterrolebinding --user system:serviceaccount:kube-system:shared-informers kube-system-cluster-admin-5 --clusterrole cluster-admin
-          '{{ kubectl }}' create clusterrolebinding --user system:kube-controller-manager  kube-system-cluster-admin-6 --clusterrole cluster-admin
-
-          # Run test
-          for test_case in internal external
-          do
-            test_file="examples/loadbalancers/${test_case}-http-nginx.yaml"
-            service_name="${test_case}-http-nginx-service"
-            # Delete fake floating-network-id to use the default one in cloud config
-            sed -i '/loadbalancer.openstack.org/d' "$test_file"
-            '{{ kubectl }}' create -f "$test_file"
-
-            if ! service_name="$service_name" timeout 600 bash -c '
-                while :
-                do
-                    [[ -n $({{ kubectl }} describe service "$service_name" | awk "/LoadBalancer Ingress/ {print \$3}") ]] && break
-                    sleep 1
-                done
-                '
-            then
-                echo "Timed out to wait for $test_case loadbalancer services deployment!"
-                '{{ kubectl }}' describe pods
-                '{{ kubectl }}' describe services
-                exit 1
-            fi
+          echo $fip
+      register: floating_ip
+      environment: '{{ global_env }}'
 
-            ingress_ip=$('{{ kubectl }}' describe service "$service_name" | awk "/LoadBalancer Ingress/ {print \$3}")
-            if curl --retry 5 --retry-max-time 30 "http://$ingress_ip" | grep 'Welcome to nginx'
-            then
-                echo "$test_case lb services launched sucessfully!"
-            else
-                echo "$test_case lb services launched failed!"
-                exit 1
-            fi
-          done
+    - name: Run acceptance tests for LoadBalancer Service
+      shell:
         executable: /bin/bash
-        chdir: '{{ k8s_os_provider_src_dir }}'
+        cmd: |
+          FLOATING_IP={{ floating_ip.stdout }} CLUSTERNAME=c886a618-c92a-4d4d-9f8f-bd2201f4dafd bash test-lb-service-catalyst-cloud.sh
```

Comment:
test-lb-service-catalyst-cloud.sh This file I upload to the openlab-zuul-jobs/tools dir, For simplest, we can just download from there.

Reply:
I put the script into cloud-provider-openstack repo, it's more reasonable to maintain this script there. PR https://github.com/kubernetes/cloud-provider-openstack/pull/934